planner/core/preprocess.go

```diff
@@ -157,6 +170,7 @@ type preprocessor struct {
 
 	// values that may be returned
 	*PreprocessorReturn
+	*PreprocessExecuteISUpdate
```

Comment:
Why not name it Preprocessor like other members?

Reply:
Why is `PreprocessExecuteISUpdate` necessary? How about just call `IsExecuteForUpdateRead` in `ensureInfoSchema`?

Reply:
import circle problem, pass function poninter here is trying to do this